src/ucs/sys/sock.c

```diff
@@ -339,10 +342,13 @@ ucs_socket_do_io_nb(int fd, void *data, size_t *length_p,
 {
     ssize_t ret;
 
-    ucs_assert(*length_p > 0);
-
     ret = io_func(fd, data, *length_p, MSG_NOSIGNAL);
-    if (ucs_likely(ret > 0)) {
+    if (ucs_likely(/* The IO operation is considered as successful if: */
+                   /* - the return value > 0 */
+                   (ret > 0) ||
+                   /* - the user's data length == 0 (Datagram sockets
+                    *   permit zero-length datagrams) */
+                   (*length_p == 0))) {
```

Comment:
why don't handle it just like ret >=0?

Reply:
what would be the value of 'ret' if (*length_p == 0) ?

Reply:
would not it be more safe to check `((*length_p == 0) && (ret==0))`?
what if length is 0 and `ret` is -1?